### PR TITLE
GCP subnet share conditions not working correctly

### PR DESCRIPTION
**Pull request by kopachevsky**
Issue  #152

Ready for review

### CODE REVIEW

**Comment by aaron-lane** [comment]
Fixes #152.

File: modules/core_project_factory/main.tf [hunk 1]
```diff
@@ -70,6 +70,28 @@ EOD
...
+resource "null_resource" "shared_vpc_subnet_invalid_name" {
+  count = "${length(var.shared_vpc_subnets)}"
+
+  triggers = {
+    name = "${replace(var.shared_vpc_subnets[count.index],
+    "/(https://www.googleapis.com/compute/v1/)?projects/[a-z0-9-]+/regions/[a-z0-9-]+/subnetworks/[a-z0-9-]+/", "") }"
+  }
+}
+
+resource "null_resource" "check_if_shared_vpc_subnets_contais_items_with_invalid_name" {
```
Comment (aaron-lane) [inline modules/core_project_factory/main.tf]:
```suggestion
resource "null_resource" "check_if_shared_vpc_subnets_contains_items_with_invalid_name" {
```

File: test/integration/full/controls/shared-vpc.rb [hunk 2]
```diff
@@ -88,28 +88,41 @@
...
+      "name" => shared_vpc_subnet_name_01,
+      "region" => shared_vpc_subnet_region_01
+    },
+    {
+      "name" => shared_vpc_subnet_name_01,
+      "region" => shared_vpc_subnet_region_01
+    }
+  ]
+
+  SHARED_VPC_SUBNETS_DATA_SET.each do |subnet|
```
Comment (aaron-lane) [inline test/integration/full/controls/shared-vpc.rb]: Test cases must not be generated through metaprogramming; this practice can lead to false positives if the logic driving the generation happens to change.

Comment (kopachevsky) [reply test/integration/full/controls/shared-vpc.rb]: I agree, but how would you minimize code duplication in this case? basically we need to do same code against 2 different networks

Comment (aaron-lane) [reply test/integration/full/controls/shared-vpc.rb]: Reducing code duplication in test cases is less of a priority than creating clear test cases. Duplicating logic in this scenario is totally acceptable.

**Comment by kopachevsky** [comment]
@aaron-lane 
> Can you please add a CHANGELOG entry and provide a [gist](https://gist.github.com) which contains the output of a Terraform process when it fails due to an invalid subnet being passed to `shared_vpc_subnets`?

Added CHANGELOG
Regarding gist, it can take to much time execute locally full test without G Suite, so I've added bad sunet name in previous commit and got:
       
```
       1 error(s) occurred:
       
       * module.project-factory.module.project-factory.null_resource.check_if_shared_vpc_subnets_contains_items_with_invalid_name: Error running command 'false': exit status 1. Output: 
```
       
on Concourse  build 422, should I copy/paste full terraform output from there for Full test suite?  I think it could contain sensitive info

**Pull request merged**